app/scripts/lib/accounts/BalancesController.ts

```diff
@@ -193,6 +200,18 @@ export class BalancesController extends BaseController<
     this.#tracker.stop();
   }
 
+  /**
+   * Gets the block time for a given account.
+   *
+   * @param account - The account to get the block time for.
+   * @returns The block time for the account.
+   */
+  #getBlockTimeFor(account: InternalAccount): number {
+    return balanceCheckIntervals[
+      account.type as BtcAccountType.P2wpkh | SolAccountType.DataAccount
+    ];
```

Comment:
This is not really future-proof IMO, whether we `throw` something here, but we should not propagate `undefined` here.

And using a default value feels kinda wrong here too.

We could follow the same pattern than `#getScopeFrom` and throw IMO.

Reply:
Will add to it